filters/include/pcl/filters/impl/crop_hull.hpp

```diff
@@ -104,81 +86,91 @@ pcl::CropHull<PointT>::getHullCloudRange ()
     -std::numeric_limits<float>::max (),
     -std::numeric_limits<float>::max ()
   );
-  for (std::size_t index = 0; index < indices_->size (); index++)
+  for (const index_t idx : *indices_)
   {
-    Eigen::Vector3f pt = input_->points[(*indices_)[index]].getVector3fMap ();
+    Eigen::Vector3f pt = input_->points[idx].getVector3fMap ();
     for (int i = 0; i < 3; i++)
     {
       if (pt[i] < cloud_min[i]) cloud_min[i] = pt[i];
       if (pt[i] > cloud_max[i]) cloud_max[i] = pt[i];
```

Comment:
Using `else if` here could avoid a bunch of comparisons but if the comparison is inexpensive I don't know whether it produces better code.

Reply:
```cpp
std::tie (cloud_min[i], cloud_max[i]) = std::minmax({cloud_min[i], cloud_max[i], pt[i]});
```
😉

Reply:
I'm not sure that `std::minmax` reduces the number of performed operations here though: it doesn't know that `cloud_min[i] <= cloud_max[i]` while my proposed use of `else if` was a direct use of that additional knowledge.

Reply:
It doesn't. Hence the wink (since now all operations are hidden)

Reply:
Oh, sorry, I totally missed the sarcasm ^^'